File: model/src/main/java/org/bandrsoftwares/celestialdiary/model/dto/establishment/AdvancedEstablishmentEquipmentContainerDTO.java
package org.bandrsoftwares.celestialdiary.model.dto.establishment;

import org.bandrsoftwares.celestialdiary.model.dto.equipment.EquipmentDTO;

import java.util.List;

public record AdvancedEstablishmentEquipmentContainerDTO(EquipmentDTO equipment, List<EstablishmentEquipmentDTO> establishmentEquipments) {
}

File: ui-frontend/src/data/model/establishment/advanced-establishment-equipment-container-dto.ts
import {EquipmentDTO} from "../equipment/equipment-dto";
import {EstablishmentEquipmentDTO} from "./establishment-equipment-dto";

export interface AdvancedEstablishmentEquipmentContainerDTO {
  equipment: EquipmentDTO,
  establishmentEquipments: EstablishmentEquipmentDTO[]
}
